Cdm web backend

Nikolay56615 opened the pull request:
Made spring initial project for backend.

Nikolay56615 committed:
Add main directory for backend

Nikolay56615 committed:
Rename package

Intelix8996 commented on cdm-web/build.gradle:
```diff
@@ -0,0 +1,44 @@
+    id 'io.spring.dependency-management' version '1.1.6'
+}
+
+group = 'ru.nsu.cdm'
```
Вот тут надо группу ещё поменять

Nikolay56615 committed:
Change group name in build.gradle

Nikolay56615 committed:
Add automatic shutdown and removal of Docker containers on application termination

dromankin committed:
Add new dependencies and user and role classes

dromankin committed:
Add main registration logic without view (needs to be tested)

- New User and Role classes
- New build and application properties
- UserRepository, UserService and RoleRepository
- MVC and security configs
- Admin and registration controllers

dromankin committed:
Fix bad DB initialization

dromankin committed:
Fix basic DB and registration issues

- Add connection to container DB
- Fix broken POST request
- Add simple login HTML page

Nikolay56615 committed:
Add lifecycle management

Nikolay56615 committed:
Merge branch 'cdm-web-docker' into cdm-web-backend

# Conflicts:
#	cdm-web/compose.yaml

dromankin committed:
Fix redirect loop

- Change GetMapping URL in DockerController
- Disable CSRF and CORS

dromankin committed:
Add password in application.properties

dromankin committed:
Add environment variable:

- .env to gitignore
- Environment variables to compose.yaml and application.properties
- Add a guide for launching application

dromankin committed:
Add readme

dromankin committed:
Change font size in readme

dromankin committed:
Fix readme

Nikolay56615 committed:
Add volumes to the docker containers

Nikolay56615 committed:
Change guide for Windows users

Nikolay56615 committed:
Change hostPort managment

Nikolay56615 committed:
Add some features to the guide

Nikolay56615 committed:
Change lifecycle management

KirShelestov committed:
Refactor connection with front-end part

dromankin committed:
Merge branch 'cdm-web-backend' of https://github.com/cdm-processors/cdm-devkit-web into cdm-web-backend

babadzakich committed:
Create Dockerfile for backend, move compose from backend folder to root

babadzakich committed:
Implement migrations to database

babadzakich committed:
Implement goose for migrations and make sql idempotent

babadzakich committed:
Fix creating containers problem

babadzakich committed:
Merge pull request #29 from cdm-processors/deploy

Implement deployment for backend

Nikolay56615 committed:
Improve frontend-backend connection

dromankin committed:
Merge conflict

dromankin committed:
Add email verification

dromankin committed:
Add bad response for registration controller

KirShelestov committed:
Fix conflict

KirShelestov committed:
Fix conflicts

dromankin committed:
Add e2e, integration and unit tests
